### ISSUE TITLE
Couldn't connect to Docker daemon at http+unix://var/run/docker.sock - is it running?

### ISSUE THREAD

**Issue by rept** [docker/compose#1214]
I tried everything in this thread: https://github.com/docker/compose/issues/88

Still no luck.

    rept@temponia:~/passenger$ netstat -ant  |grep 4243
    tcp        0      0 127.0.0.1:4243          0.0.0.0:*               LISTEN     
    rept@temponia:~/passenger$ echo $DOCKER_HOST
    tcp://localhost:4243
    rept@temponia:~/passenger$ sudo docker-compose build db uses an image, skipping
    Building web...
    Couldn't connect to Docker daemon at http+unix://var/run/docker.sock - is it running?

    If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.

This is the version:

    rept@temponia:~/passenger$ sudo docker version
    Client version: 1.0.1
    Client API version: 1.12
    Go version (client): go1.2.1
    Git commit (client): 990021a
    Server version: 1.0.1
    Server API version: 1.12
    Go version (server): go1.2.1
    Git commit (server): 990021a

Any ideas?  I also tried installing fig which works right away without any problem...

**Comment by leoyuholo** [comment]
I get the same error message when I run docker.
It is solved by adding myself to the user group 'docker'

Try to run the following command to add yourself to the group
`usermod -aG docker ${USER}`

Or you can run the following command to find out what groups you belong to
`groups $USER`

**Comment by thaJeztah** [comment]
Looks like docker-compose is disregarding the `$DOCKER_HOST` env var here, because it's trying to connect to the socket in stead of the 127.0.0.1 address.

**Comment by kekko1212** [comment]
same issue here

**Comment by tylerteh** [comment]
I encountered same error message just now, that time my laptop WiFi is off without wired connection. I enable WiFi and connected to home wireless network then problem solved.

**Comment by comxd** [comment]
You can try the 5.0.3 (prebuild) at https://www.virtualbox.org/wiki/Testbuilds ? It's fix this issue for me.

**Comment by magnunve** [comment]
i had that problem using docker in my mac, but it got solved one i ran the compose command without sudo, so it seems it is a user issue,

**Comment by vkeziah** [comment]
Soleve for me too with sudo command

**Comment by thaJeztah** [comment]
@rept are you still having this issue? The docker version you reported is over a year old now, which is, well, pre-historic in docker terms :)

I wonder if this issue should be kept open because it's now collecting all kind of *possible* reasons the daemon cannot be reached (I don't mean this in a bad way; thanks everyone for suggesting solutions, it is appreciated)

**Comment by rept** [comment]
I did a new installation on a completely different machine and it's working now. So you van close it.

**Comment by thaJeztah** [comment]
Thanks, @rept, and good to hear it was resolved!

**Comment by alexanderadam** [comment]
I get

```bash
ERROR: Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running?
```
too everytime I activate [`tty: true` or `stdin_open: true`](https://docs.docker.com/v1.8/compose/yml/#working-dir-entrypoint-user-hostname-domainname-mac-address-mem-limit-memswap-limit-privileged-restart-stdin-open-tty-cpu-shares-cpuset-read-only-volume-driver) in `docker-compose.yml` and running `docker-compose up`.

Can it be related?
```
Docker version 1.9.1, build a34a1d5
docker-compose version 1.5.2, build 7240ff3
```

**Comment by romeromfm** [comment]
I just solved my problem running the "docker-compose run ..." command with sudo.

**Comment by jurajseffer** [comment]
I ran into this when running `docker compose up -d` during vagrant up. User `vagrant` has been added to group `docker` but the above error appears when running it without `sudo`. It works later on if one logs in as user `vagrant` but fails during provisioning so it has something to do with env variables?

`docker-compose version: 1.4.1`

**Comment by cloojure** [comment]
Simple solution:      **sudo docker-compose up**

**Comment by immarvin** [comment]
I ran to same issue with:
```
[root@c910f05c01bc07 xcat-docker]# docker-compose version
docker-compose version 1.6.2, build 4d72027
docker-py version: 1.7.2
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013
```

**Comment by mashimom** [comment]
That is a workaround not a solution, right?

**Comment by sergeyklay** [comment]
I still often face with this

**Comment by dnephin** [comment]
This is not a compose issue. If you can't connect to the docker daemon from compose there are a few likely causes:
* you forgot to export the `DOCKER_` env variables, which is often done by running `eval $(docker-machine env default)`
* there is a network problem between you and the remote docker host (if you're running on a cloud provider)
* there is a problem with the docker daemon running in the Virtualbox VM
* your user doesn't have access to the docker daemon socket because of permissions (if you're running on a linux host). The solution is to sudo or fix the permissions.

These aren't problems that can be solved by compose.

There is an issue with `tty`, which is ticketed here: #3106

**Comment by madflow** [comment]
I can confirm: This is not a docker-compose issue.

If it helps: I came across the same issue (Couldn't connect...) on a Ubuntu host. Running ```sudo systemctl status docker.service -l ``` showed the "real" issue. The solution in my case was to remove all files in "/var/lib/docker/network/files/" and restart the docker service.

**Comment by immarvin** [comment]
hi @dnephin @madflow , I think this is a docker-compose issue.

Since with the same docker image and docker network, the container can be started up successfully with
```
 sudo docker run -it --privileged=true  --hostname=xcatmn --name=xcatmn --add-host="xcatmn.clusers.com xcatmn:10.5.107.101" --volume /docker/xcatdata/:/install --net=subnet1 --ip=10.5.107.101  xcat/xcat-ubuntu-x86_64
```

but failed with "docker-compose up". The docker-compose.yml is right the equivalent of the options and arguments of "docker run":
```
[root@c910f05c01bc07 xcat-docker]# docker-compose config
networks:
  default:
    external:
      name: subnet1
    external_name: subnet1
services:
  xcatmn:
    container_name: xcatmn
    extra_hosts:
      xcatmn.clusers.com xcatmn: 10.5.107.101
    hostname: xcatmn
    image: xcat/xcat-ubuntu-x86_64
    networks:
      default:
        ipv4_address: 10.5.107.101
    privileged: true
    stdin_open: true
    tty: true
    volumes:
    - /docker/xcatdata:/install:rw
version: '2.0'
volumes: {}
```

The docker and compose version info:
```
[root@c910f05c01bc07 xcat-docker]# docker version
Client:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   c3959b1
 Built:        Mon Feb 22 16:16:33 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   c3959b1
 Built:        Mon Feb 22 16:16:33 2016
 OS/Arch:      linux/amd64
[root@c910f05c01bc07 xcat-docker]# docker-compose version
docker-compose version 1.7.0dev, build 99d68be
docker-py version: 1.8.0-dev
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013
```

**Comment by dnephin** [comment]
If you can reproduce a connection issue with Compose and not with docker cli please open a new issue. Including all the information you have here is great, and also the error message, and a link to a paste with `docker-compose --verbose up`.

**Comment by chriskessel** [comment]
I get this if any of my container names have camel casing. This fails with that message:

```
referenceCombo:
  build: referenceCombo

Building referencCombo
ERROR: Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running?

If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.
```

Change the name to "referencecombo" (all lowercase) and it's fine.  The error message definitely doesn't help figure this one out :(

**Comment by dnephin** [comment]
@chriskessel I think your issue is covered by #2816, it's a bug in Docker Engine.

**Comment by vimalprakashts** [comment]
after restart docker services , it works for me :( 

> service docker restart

**Comment by Mahmoudz** [comment]
I had the same error, after 15 min of debugging. Turns out all it needs is a `sudo` :) 
Check out [Create a Docker group] here https://docs.docker.com/engine/installation/linux/ubuntulinux/ to get rid of the sudo prefix.

**Comment by fabiopaiva** [comment]
Why needs to run with sudo?

**Comment by dpetzold** [comment]
@fabiopaiva by default docker connects to a unix socket which is a file with root user and docker group read/write privileges:

  $ ls -l /var/run/docker.sock
  srw-rw---- 1 root docker 0 Jun 29 23:38 /var/run/docker.sock

so you either need to be root or a member of the docker group to connect to it. If docker was listening on a tcp socket the permissions would not apply.

**Comment by pixies** [comment]
I delete file /var/run/docker.sock
how could correct?

**Comment by thaJeztah** [comment]
@pixies restart the docker service should recreate it

**Comment by marcus233** [comment]
This is the correct answer. In many cases you will need to log out and log back in for the change to take effect. Running the command with sudo will work too, but adding the necessary users to the docker group is a better solution so you don't have to give out sudo access.

leoyuholo commented on May 14, 2015 
> It is solved by adding myself to the user group 'docker'
> 
> Try to run the following command to add yourself to the group
> `usermod -aG docker ${USER}`
> 
> Or you can run the following command to find out what groups you belong to
> `groups $USER`

**Comment by ajahongir** [comment]
it seams really work if restart a machine.

**Comment by CnytSntrk** [comment]
@alexanderadam  did u find any solution ? i got the same error now

**Comment by alexanderadam** [comment]
@CnytSntrk sorry, I solved the problem otherwise. :wink: 
But good luck :four_leaf_clover: finding a solution!

**Comment by CnytSntrk** [comment]
@alexanderadam thanks :+1:

**Comment by arshpreetsingh** [comment]
I just solved my problem running the "docker-compose run ..." command with sudo.

**Comment by bakeris** [comment]
I solved the problem adding sudo

**Comment by cuipengfei** [comment]
rebooting did the trick for me

**Comment by rahman6848** [comment]
after adding user to docker group reboot is not required, just log out and login, its working for me

**Comment by valeriaelias** [comment]
I tryed before rahman. It didn't work. Thanks anyway

**Comment by caneraydinbey** [comment]
+1

**Comment by vadi2** [comment]
This is still an issue on Ubuntu 16.04 and adding myself to the docker group, then relogging worked. This issue should not be closed.

**Comment by colinpade** [comment]
just hit this in Ubuntu 16.04

**Comment by ytokan** [comment]
Ubuntu 16.04 
https://github.com/docker/compose/issues/1214#issuecomment-102246925 
Work perfect but restart before

**Comment by nejeoui** [comment]
Ubuntu 16.04  has the same error

**Comment by mikevoets** [comment]
Solution (from https://docs.docker.com/engine/installation/linux/debian/, does not only work with Debian):

# Add the docker group if it doesn't already exist.
$ sudo groupadd docker

# Add the connected user "${USER}" to the docker group.
# Change the user name to match your preferred user.
# You may have to logout and log back in again for
# this to take effect.
$ sudo gpasswd -a ${USER} docker

# Restart the Docker daemon.
$ sudo service docker restart

**Comment by kirkins** [comment]
@mikevoets yes I think that should solve this issue. 

I had the same thing happen on a new install and realized it was because I wasn't using 'sudo'.

**Comment by andkirby** [comment]
@rept it doesn't work any more if it worked before. Adding a user into the group doesn't work.

**Comment by mordonez-me** [comment]
Does anyone knows how to run it without using sudo?

**Comment by mikevoets** [comment]
See my reply Marco

2016-11-08 0:15 GMT+01:00 Marco Ordonez <notifications@github.com>:

**Comment by andkirby** [comment]
@mordonez-me, adding into docker group will help.

`sudo gpasswd -a ${USER} docker`

**Comment by pbardondev** [comment]
this only worked for me after logging in and out after adding my user to the docker group.

**Comment by uberhacker** [comment]
FWIW, I needed a system restart before it started working for me.  This was likely due to the fact the `linux-image-extra-*` kernel packages were freshly installed as required by `docker` on Ubuntu.  See https://docs.docker.com/engine/installation/linux/ubuntulinux/.

**Comment by harshajayaweeraXHJ** [comment]
Try running "dockerd" or "sudo dockerd" if required

**Comment by czerasz** [comment]
Since I used some of the suggested commands in this thread, I wanted to share my experiences.

I have the following configuration in the `Vagrantfile`:

```
config.vm.provision "docker"
config.vm.provision "shell", inline: "docker ps", privileged: false
# or
# config.vm.provision "shell", inline: "docker-compose ps", privileged: false
```

I get the following issue:

- when I try to use Docker:

  ```
  Cannot connect to the Docker daemon. Is the docker daemon running on this host?
  ```

- when I try to use Docker Compose:

  ```
  ==> default: Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running?
  ==> default: 
  ==> default: If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.
  ```

Thanks to [this comment](https://github.com/mitchellh/vagrant/issues/3998#issuecomment-60359659) I got it working.

Solution:

```
config.vm.provision "docker"
# The following line terminates all ssh connections. Therefore
# Vagrant will be forced to reconnect.
# That's a workaround to have the docker running with the vagrant user
config.vm.provision "shell", inline:
   "ps aux | grep 'sshd:' | awk '{print $2}' | xargs kill", privileged: true
config.vm.provision "shell", inline: "docker ps", privileged: false
```

**Comment by ebuildy** [comment]
this is not a Docker related issue, but more a linux socket permission problem (very common issue between php-fpm and nginx).

1) Docker is a classic HTTP client/server application, the client will speak via a socket or IP to the server
2) The socket is a linux file, with permission, it belongs to an user and a group:

```
    sudo ls -alh /var/run
    ...
    srw-rw---- 1 root staff    0 Jan 22 10:09 docker.sock
    ....
```

3) When doing ``docker ps``,  you must have permission to access to ``docker.sock`` file

In this case, just do ``sudo usermod -aG staff [ME]`` to add you in the right group (usually it's ``docker``) (same thing with php-fpm, where the group is usually www-data).

**Comment by oesah** [comment]
I had the problem, that I used `pip install docker-compose`. I uninstalled it and used `sudo pip install docker-compose`. Now the command `sudo docker-compose up` works. 

Hope that helps.

**Comment by dagisus** [comment]
whenever you have authentication problems with a pip install is best to use pip install [package] —user than using sudo, in some cases using sudo will cause authentication problems while executing the application itself.

**Comment by jgawlik83** [comment]
on Fedora Linux
add your user to docker group
`sudo usermod -aG docker username`

after this command **reboot**, logout is not enough as on other distros

**Comment by cinqs** [comment]
Nothing is better than recommending the official doc:

https://docs.docker.com/engine/installation/linux/linux-postinstall/

check this out.

The official doc is always the first place where you should go for help.

**Comment by digitaldreamer** [comment]
So I ended up adding the user to the docker group:

```
sudo groupadd docker
sudo usermod -aG docker $USER
sudo service docker restart
```

I still had connection issues when running `docker ps` as the user giving me the following error:

```
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.27/containers/json: dial unix /var/run/docker.sock: connect: permission denied
```

Rebooting the server fixed this issue allowing me to run docker commands as the user without using sudo:

```
sudo reboot now
... (wait for boot) ...
ssh server
docker ps
```

**Comment by fkaralis** [comment]
using sudo solved it for me too

**Comment by faost** [comment]
Without logout:

```
sudo usermod -aG docker $USER
su $USER -
```

**Comment by TrkiSF2** [comment]
As mentioned above indeed: 

`sudo usermod -aG docker $USER`

**Logout.**
Login.
Done.

**Comment by finalclass** [comment]
I had the same problem and in my case the cause was that I had some files in the folder with Dokcerfile to which I had no access rights. After running `chown` all started to work nice.
The solution with `usermod` hasn't been working for me.

**Comment by WandyLau** [comment]
This is actually a problem of installation. Please refer to the official page the for post-installation.  I solve this problem with the help of the document.

**Comment by davidchappelle** [comment]
As @TrkiSF2 said above. After changing the group settings you need to logout and log back in again for them to take effect.

**Comment by angelovaldez** [comment]
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.29/containers/json: dial unix /var/run/docker.sock: connect: permission denied

solution: # chmod -Rf 777 /var/run/docker.sock

**Comment by caneraydinbey** [comment]
Restarting solved after doing 

> sudo usermod -aG docker ${USER}

**Comment by aaksarin** [comment]
Got this problem on digital ocean droplet. Setting droplet to internal kernel management fixed this problem for me https://www.digitalocean.com/community/tutorials/how-to-update-a-digitalocean-server-s-kernel#setting-up-the-droplet-for-internal-kernel-management

**Comment by mariano-e** [comment]
Doing this, solve my problem:
https://techoverflow.net/2017/03/01/solving-docker-permission-denied-while-trying-to-connect-to-the-docker-daemon-socket/

**Comment by ChrisPappalardo** [comment]
I had this problem on a code repo that was building fine one moment and then not.  I had other repos on the same machine and same account that built just fine, so it wasn't a groups issue or anything with the system configuration.  I had no DOCKER_HOST env var set.  I suspect this error can be caused by a variety of reasons, but for me it ended up being root ownership of the build artifacts in the repo directory.  `sudo chown -R` all files in the repo to the user account solved the problem for me.

**Comment by franklingu** [comment]
I personally tried @cloojure 's solution and it worked. But only after I added myself to docker group and re-login. The re-login part is essentail

**Comment by dgendill** [comment]
I ran into this same error message, and just like [ChrisPappalardo](https://github.com/docker/compose/issues/1214#issuecomment-308859968) it was due to a file permission issue not the daemon. `sudo chown -R` on the build assets fixed it.

**Comment by Calliari** [comment]
I get the same error message when I run "docker-compose build".
So I added the sudo command and it's works fine. ($ sudo docker-compose build)

**Comment by Farada** [comment]
this might be another solution:
`sudo chmod 777 /var/run/docker.sock`

**Comment by thaJeztah** [comment]
@Farada changing the permissions is dangerous; having access to the Docker API is equivalent to having root access on your machine; changing the permissions to 777 gives anyone on the host root access.

**Comment by kevinhowbrook** [comment]
Adding your user to the docker group is the solution for this, as mentioned above you need to reboot your server/computer. Please don't chmod anything to 777!

**Comment by Daniel900318** [comment]
If you installed docker using sudo command, try sudo YOUR_COMMAND.

**Comment by andersevenrud** [comment]
In my case I had a hidden directory with `root:root` permissions in my build directory. I removed this folder and `docker-compose build` now works.

This folder came from one of the images I'm using, which is not created until I run `docker-compose up`. So the next time I try to build I get this error.

So basically, I was given an unrelated error.

**Comment by wyqydsyq** [comment]
I've been getting this:
```
 docker-compose build
db uses an image, skipping
Building web
ERROR: Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running?

If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.
```
The issue was that some files in the context of the Dockerfile had restrictive permissions, a quick `sudo chown -R damo ./*` fixed it.

However I would argue there is indeed an issue with `docker-compose` here — I should have received an informative error message that accurately specifies that it failed because of a permissions error, rather than an ambiguous connection error which was a red herring in my case as I wasted time messing with the docker daemon service itself.

**Comment by augnustin** [comment]
I got this few times.

For me it is no config issue. Instead, it feels like Docker dies and stop, during some event I still can't tell *when* it occurs.

Anyway if my intuition is correct, the fix could be `sudo service docker restart` but apparently this is not sufficient.

So at the end the only solution that works is **a reboot**.

**Comment by renatoagomes** [comment]
> The docker daemon binds to a Unix socket instead of a TCP port. By default that Unix socket is owned by the user root and other users can only access it using sudo. The docker daemon always runs as the root user. [Linux Post Install ](https://docs.docker.com/engine/installation/linux/linux-postinstall/)

```
$ sudo groupadd docker
$ sudo usermod -aG docker $USER
```

**reminder:** (remember to restart session or reboot VM )

- If testing on a virtual machine, it may be necessary to restart the virtual machine for changes to take affect.
- On a desktop Linux environment such as X Windows, log out of your session completely and then log back in.

**Comment by dineshdharme** [comment]
I had the same problem. As others have mentioned, when I ran the docker-compose with sudo, it worked.

**Comment by vzool** [comment]
Hi, guys I faced the same issue while I'm using Fedora 26 and then I found the solution for Fedora 25, but it does work for me.

Here is the snippet:
```bash
$ sudo groupadd docker && sudo gpasswd -a ${USER} docker && sudo systemctl restart docker
$ newgrp docker
```
REF: [https://developer.fedoraproject.org/tools/docker/docker-installation.html](https://developer.fedoraproject.org/tools/docker/docker-installation.html)

I hope it help you, Thanks

**Comment by GCorbel** [comment]
I had the same but only with `docker-compose build`. This was finally just a problem of file permission. I created files inside the container, owned by root, and it was not able to build with those files. I simply did a `sudo chown -R $USER .`.

**Comment by rishadomar** [comment]
I was having the same error creating a mysql container after having deleted my previous one.
The reason was that I had a data folder (for persistence) that was owned by the user in the previous container.
The solution was to delete the data folder. Warning - you will lose your data. If you wish to keep your data maybe move the data folder or assign new owners.

**Comment by rhafiko** [comment]
did the trick adding myself to the group
usermod -aG docker ${USER}

reboot and works !

**Comment by Arch88** [comment]
For me this was resolved by login to admin mode :  sudo -s 

If this helps !

**Comment by tcorbettclark** [comment]
I obtained this same error message by using an image name containing a "+" character. (This came about because I was naming my images after a version control id.)

For example, a `docker-compose.yaml` containing:

```
services:
  foobar:
    image: "foobar:a-b+c"
    ....

```
fails with

```
Building foobar
ERROR: Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running?

If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.

```
But removing the "+" from the image name worked fine.

**Comment by Gronan** [comment]
@fabiopaiva Not sure if someone replyed to you but it needs to be sudo exclusive at first because any user that can execute the docker command can become sudo **in the host**.
Anyone with docker execution access can launch a simple docker bind the entire host filesystem and access it as root (i.e shadow files for passwords or modify permission for the entire system).
So you have to **be very carefull** who can use this command.

**Comment by fabiopaiva** [comment]
Thanks @Gronan, I don't remember clearly about this thread.
But, regarding your explanation, if you run with sudo you give this control anyway.
Thank you for the explanation

**Comment by iahmad-khan** [comment]
Dont need sudo. 
Add your user to the docker group
and restart the docker service

**Comment by bluemix** [comment]
` usermod -aG docker ${USER}`
then _reboot_.

**Comment by wjsc** [comment]
This is because the **docker** daemon runs as the root user nnd  **docker-compose** as a regular user

**Three steps to fix it:** 
1. sudo groupadd docker
2. sudo usermod -aG docker $USER
3. And reboot to refresh the changes.

More details in: 
https://docs.docker.com/engine/installation/linux/linux-postinstall/#manage-docker-as-a-non-root-user

**Comment by edwardboyle** [comment]
On AWS Ubuntu

`sudo usermod -aG docker $USER`

- End SSH Session
- Re-Connect

**Comment by niklasravnsborg** [comment]
Instead of rebooting you can just type `su - ${USER}`. I guess this reloads the currently active user in the Terminal. 

https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-16-04

**Comment by rbahumi** [comment]
This worked for me without the reboot:
1. Add docker group:
`sudo usermod -aG docker ${USER}`
2. Run
`id`
3. If the 'docker' group does not appear in the group list - logout and login again

**Comment by alexcdot** [comment]
My problem ended up being that in my docker build context, I had files that were neither owned by muser 
 yor group. For example, I mounted a folder in my docker build context to a mongo container, so these files were owned by the mongodb user and group.

**Comment by tbuss93** [comment]
I can confirm @alexcdot observation: Had files in my project repository owned by root (dunno how they got there). Deleting them resolved the issue. Now I can run docker-compose without sudo privileges.

**Comment by FrostyX** [comment]
Also confirming the @alexcdot and @tbuss93 solution. Worked for me as well. I think that in such case, the error should be more user-friendly and ideally show problematic files.

**Comment by cooljl31** [comment]
i had this issue today for a new Dev notebook. I solved it like this
1) Stop Docker
```
sudo service docker stop
```
if you look at this file permission you'll see that without `sudo` you won't be able to get access to Docker daemon
```
sudo ls -la /var/run/docker.sock
srw-rw---- 1 root docker 0 Dec 21 19:16 /var/run/docker.sock
```
2) I remove this docker.sock
```
sudo rm  /var/run/docker.sock
```
thanks @rbahumi 
3) i run this command
```
sudo usermod -aG docker ${USER}
```
and now you are almost done just Log out or better just restart the computer you don't have to start the service because it start automatically

**Comment by tbuss93** [comment]
@cooljl31 removing the socket seems to be unnecessary, as the permissions show that members of the docker group have rw access. You will have to restart docker and the session though.

This fix, however, seems to be unrelated to the issue

**Comment by CajuCLC** [comment]
This solved my problems: https://docs.docker.com/install/linux/linux-postinstall/#manage-docker-as-a-non-root-user
Just remember to log out and log back in. Run the commands as the user you want to use docker.

**Comment by hassamali** [comment]
sudo usermod -aG username

Worked like a charm for me.

**Comment by AydinChavez** [comment]
I had to also start the docker service manually after a new installation of docker-ce (17.12.0~ce-0~ubuntu)  and docker-compose to make the error message disappear:

`sudo service docker start `

**Comment by matchilling** [comment]
Following the steps described in the [documentation](https://docs.docker.com/install/linux/linux-postinstall/) resolved the issue with my new Ubuntu 16.04 installation.

**I had to reboot the system after applying those changes though.**

**Comment by dhirajpatra** [comment]
Use command:   `sudo docker-compose up -d`

**Comment by derekng622** [comment]
if u met:
`$ docker-compose up
ERROR: Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running?

If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.`
u should check if current login user belongs to `/var/run/docker.sock` user group `docker`.
Try run Compose as sudo.

**Comment by averissimo** [comment]
To solve this without `sudo` do as @leoyuholo said in the second comment and logout and back in.

>I get the same error message when I run docker.
>It is solved by adding myself to the user group 'docker'
>
>Try to run the following command to add yourself to the group usermod -aG docker ${USER}
>
>Or you can run the following command to find out what groups you belong to groups $USER

After that logout and log back in and it works

link to comment https://github.com/docker/compose/issues/1214#issuecomment-102246925

**Comment by abepetrillo** [comment]
Just because it takes my a long time to re-dsicover this, and is the first thing I hit on google. There is nothing wrong with my user permissions on my laptop. The problem was the file permissions on one of the mounted folders. If you either fix the permissions so you don't need root to access them, or delete them (in my case it is cached assets for rails) it successfully goes through.

The real problem is the error message does not tell you what the specific problem is when you build via docker-compose.

**Comment by zjhken** [comment]
I met the issue also. It is a user access issue.
I solved it by install docker-compose in root mode and run docker-compose also in root mode.

**Comment by tbuss93** [comment]
@zjhken that is not really a solution, but rather a workaround.
There is something wrong with either the file permissions in your project folder or your group permissions. See @alexcdot and my observations for the first case, and try `sudo usermod -aG docker <your username>` followed by re-logging-in for the latter.

**Comment by zjhken** [comment]
@tbuss93 you're right. I have use you solution to solve it. It works. This might be the best solution.

**Comment by diptoo** [comment]
write =>  sudo docker info

**Comment by HargovindArora** [comment]
Just add `sudo` before the command which you are trying to run!

**Comment by piojoalejandro** [comment]
People, firts at all my apoligize for my english. i'm have th same problem but a realize that its happen on konsole inside kde, but when i run in a terminal crt alt f1 i dont need use sudo!. Iḿ already have $USER on docker group. I hope you ca understand me.- thanks

**Comment by alvaropaco** [comment]
Just run;

`sudo chown $USER:docker /var/run/docker.sock`

**Comment by kuhajeyan** [comment]
after adding user to docker group
restarting the machine will do.

**Comment by roshangl** [comment]
For me its working. 
[ec2-user@ip-172-31-7-179 docker-elk]$  sudo docker-compose up -d

My docker and docker compose version are.

[ec2-user@ip-172-31-7-179 docker-elk]$ sudo docker version
Client:
 Version:      17.06.2-ce
 API version:  1.30
 Go version:   go1.8.4
 Git commit:   3dfb8343b139d6342acfd9975d7f1068b5b1c3d3
 Built:        Fri Nov 10 00:50:37 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.2-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.4
 Git commit:   402dd4a/17.06.2-ce
 Built:        Fri Nov 10 00:51:08 2017
 OS/Arch:      linux/amd64
 Experimental: false
[ec2-user@ip-172-31-7-179 docker-elk]$ sudo docker-compose version
docker-compose version 1.21.2, build a133471
docker-py version: 3.4.0
CPython version: 2.7.12
OpenSSL version: OpenSSL 1.0.2k-fips  26 Jan 2017

**Comment by orapradeep** [comment]
after following add user procedure make sure u restart your machine.

**Comment by mikeyjk** [comment]
I also experience this when I have a project structured like this:

- docker-compose.yml
- Dockerfile
- volumes

If 'volumes' exceeds a certain size, I get the following error, when running 'docker-compose build':

-> ERROR: Couldn't connect to Docker daemon at http+docker://localunixsocket - is it running? 
-> If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.

Running the command with sudo 'resolves' the issue / it will begin to build the image.
My cause: not adding 'volumes' to a .dockerignore file, to prevent my binary data inside volumes being passed into the build context. It's nice to not have to rely on sudo for automation.

**Comment by praveenbaliga** [comment]
cd /var/run
sudo chmod 777 docker.sock
sudo service docker start
docker run hello-world
Hello from Docker!
This message shows that your installation appears to be working correctly.

Solved the problem :)

**Comment by bitbd83** [comment]
solved with sudo

**Comment by joao-pedro-alves** [comment]
If everything said here don't work, reinstall docker. It worked for me.

**Comment by jerryrelmore** [comment]
No need to restart the machine in my experience which is helpful if you're in a production environment - after `sudo usermod -aG docker $USER`, log out and log back in. That should pass the new user group assignment to any program or app that needs it. 

Failing that, @praveenbaliga 's response 3 comments up from my comment will work in a pinch if the permissions were incorrect to begin with or you're in a staging environment, but may violate security policies if you're working in a hardened production environment. You really only want `root` and `docker` group members to have access to the `docker.sock` and `chmod 777` will give `rwx` access to every user.

**Comment by nileshparwan** [comment]
use this link to solve the problem as it works for me. I had the same problem but now it works fine

https://techoverflow.net/2017/03/01/solving-docker-permission-denied-while-trying-to-connect-to-the-docker-daemon-socket/

**Comment by AnostDev** [comment]
> I get the same error message when I run docker.
> It is solved by adding myself to the user group 'docker'
> 
> Try to run the following command to add yourself to the group
> `usermod -aG docker ${USER}`
> 
> Or you can run the following command to find out what groups you belong to
> `groups $USER`

And then restart your computer

**Comment by leimao** [comment]
This thread is helpful.

**Comment by zhongw117** [comment]
I entered the same issue. On the contrary, I add sudo at the beginning of my command line and it solved!

**Comment by adonese** [comment]
> Just run;
> 
> `sudo chown $USER:docker /var/run/docker.sock`

This one worked for me. Not sure why adding $USER to docker group didn't solve it.

**Comment by an1rk4** [comment]
> I ran into this when running `docker compose up -d` during vagrant up. User `vagrant` has been added to group `docker` but the above error appears when running it without `sudo`. It works later on if one logs in as user `vagrant` but fails during provisioning so it has something to do with env variables?
> 
> `docker-compose version: 1.4.1`

sudo groupadd docker
sudo gpasswd -a vagrant docker // Add the current user vagrant to the docke group
sudo service  docker restart // restart docker service
